Compile fails with Java 17 due to old plugin versions - "java.lang.ExceptionInInitializerError: Cannot access defaults field of Properties"

If I do a `mvn clean package` from the top-level 'finish' directory I get an error like the below.  Can fix by upgrading plugin versions.

> [INFO] --- maven-war-plugin:2.2:war (default-war) @ guide-maven-multimodules-war ---
> [WARNING] Error injecting: org.apache.maven.plugin.war.WarMojo
> com.google.inject.ProvisionException: Unable to provision, see the following errors:
> 
> 1) Error injecting constructor, java.lang.ExceptionInInitializerError: Cannot access defaults field of Properties
>   at org.apache.maven.plugin.war.WarMojo.<init>(Unknown Source)
>   while locating org.apache.maven.plugin.war.WarMojo
> 
> 1 error
>     at com.google.inject.internal.InternalProvisionException.toProvisionException (InternalProvisionException.java:226)
>     at com.google.inject.internal.InjectorImpl$1.get (InjectorImpl.java:1053)
>     at com.google.inject.internal.InjectorImpl.getInstance (InjectorImpl.java:1086)
>     at org.eclipse.sisu.space.AbstractDeferredClass.get (AbstractDeferredClass.java:48)
>     at com.google.inject.internal.ProviderInternalFactory.provision (ProviderInternalFactory.java:85)
>     at com.google.inject.internal.InternalFactoryToInitializableAdapter.provision (InternalFactoryToInitializableAdapter.java:57)
>     at com.google.inject.internal.ProviderInternalFactory$1.call (ProviderInternalFactory.java:66)
>     at com.google.inject.internal.ProvisionListenerStackCallback$Provision.provision (ProvisionListenerStackCallback.java:112)
>     at com.google.inject.internal.ProvisionListenerStackCallback$Provision.provision (ProvisionListenerStackCallback.java:127)
>     at com.google.inject.internal.ProvisionListenerStackCallback.provision (ProvisionListenerStackCallback.java:66)
>     at com.google.inject.internal.ProviderInternalFactory.circularGet (ProviderInternalFactory.java:61)
>     at com.google.inject.internal.InternalFactoryToInitializableAdapter.get (InternalFactoryToInitializableAdapter.java:47)
>     at com.google.inject.internal.InjectorImpl$1.get (InjectorImpl.java:1050)
>     at org.eclipse.sisu.inject.Guice4$1.get (Guice4.java:162)
>     at org.eclipse.sisu.inject.LazyBeanEntry.getValue (LazyBeanEntry.java:81)
>     at org.eclipse.sisu.plexus.LazyPlexusBean.getValue (LazyPlexusBean.java:51)
>     at org.codehaus.plexus.DefaultPlexusContainer.lookup (DefaultPlexusContainer.java:263)
>     at org.codehaus.plexus.DefaultPlexusContainer.lookup (DefaultPlexusContainer.java:255)
>     at org.apache.maven.plugin.internal.DefaultMavenPluginManager.getConfiguredMojo (DefaultMavenPluginManager.java:520)
>     at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:124)
>     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
>     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
>     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
>     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
>     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
>     at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
>     at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
>     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
>     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
>     at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
>     at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
>     at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
>     at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
>     at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
>     at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
>     at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
>     at java.lang.reflect.Method.invoke (Method.java:568)
>     at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
>     at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
>     at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
>     at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
> Caused by: java.lang.ExceptionInInitializerError: Cannot access defaults field of Properties
>     at com.thoughtworks.xstream.converters.collections.PropertiesConverter.<clinit> (PropertiesConverter.java:46)
>     at com.thoughtworks.xstream.XStream.setupConverters (XStream.java:647)
>     at com.thoughtworks.xstream.XStream.<init> (XStream.java:445)
>     at com.thoughtworks.xstream.XStream.<init> (XStream.java:385)
>     at com.thoughtworks.xstream.XStream.<init> (XStream.java:342)
>     at org.apache.maven.plugin.war.util.WebappStructureSerializer.<clinit> (WebappStructureSerializer.java:47)
>     at org.apache.maven.plugin.war.AbstractWarMojo.<init> (AbstractWarMojo.java:316)
>

--- a/finish/pom.xml
+++ b/finish/pom.xml
@@ -26,13 +26,25 @@
     <!-- end::modules[] -->
 
     <build>
+        <pluginManagement>
+            <plugins>
+                <plugin>
+                    <artifactId>maven-war-plugin</artifactId>
+                    <version>3.3.2</version>
+                </plugin>
+                <plugin>
+                    <artifactId>maven-compiler-plugin</artifactId>
+                    <version>3.10.1</version>
+                </plugin>
+            </plugins>
+        </pluginManagement>
         <plugins>
             <!-- Enable liberty-maven plugin -->
             <!-- tag::liberty-maven-plugin[] -->
             <plugin>
                 <groupId>io.openliberty.tools</groupId>
                 <artifactId>liberty-maven-plugin</artifactId>
-                <version>3.5.1</version>
+                <version>3.7.1</version>
             </plugin>
             <!-- end::liberty-maven-plugin[] -->
         </plugins>

